src/averageAges.js

```diff
@@ -1,64 +1,52 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+const SEX_MALE = 'm';
+const SEX_FEMALE = 'f';
+
+function calculateTotalAge(people) {
+  return people.reduce((sum, person) => sum + (person.died - person.born), 0);
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filterMen = people.filter(({ sex, died }) =>
+    sex === SEX_MALE && (century
+      ? Math.ceil(died / 100) === century
+      : true
+    ));
+
+  const totalAgeMen = calculateTotalAge(filterMen);
+
+  return totalAgeMen / filterMen.length;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filterWomans = people.filter(({ sex, name }) =>
```

Comment:
https://github.com/mate-academy/js_average-ages/pull/2336#discussion_r1293138520
```suggestion
  const filteredWomen = people.filter(({ sex, name }) =>
```